bench/ring_memberlist_test.go

```diff
@@ -0,0 +1,96 @@
+package bench
```

Comment:
I didn't know the best place to put this. The test is specifically targeting a performance characteristic of the `kv/memberlist/` code when using the `/ring` data structures, so I do not believe it belongs in either of those. If anything, this is an integration test (of sorts, benchmark) covering the two when used together.

Reply:
👍 Moving it to either of those two would create cyclic dependency :( (`ring` package depends on `memberlist` already).

Reply:
[nit] I would just move it to `ring/bench` since it's a benchmark on ring use case. But don't feel strong about it.